executor/infoschema_reader.go

```diff
@@ -1848,12 +1848,15 @@ func (e *memtableRetriever) setDataForPlacementPolicy(ctx sessionctx.Context) er
 			continue
 		}
 		// Currently, only partitions have placement rules.
+		var tbName, dbName, ptName string
 		tb, db, part := is.FindTableByPartitionID(id)
-		if tb == nil {
-			return errors.Errorf("Can't find partition by id %d", id)
-		}
-		if checker != nil && !checker.RequestVerification(ctx.GetSessionVars().ActiveRoles, db.Name.L, tb.Meta().Name.L, "", mysql.SelectPriv) {
-			continue
+		if tb != nil {
```

Comment:
Still show the row even if the partition doesn't exist?

Reply:
It is intended for testing. We don't want a filtered list of rules if we do further tests. Let me add a failpoint.